pkg/disttask/framework/taskexecutor/slot_test.go

```diff
@@ -23,32 +23,79 @@ import (
 
 func TestSlotManager(t *testing.T) {
 	sm := slotManager{
-		executorSlotInfos: make(map[int64]*slotInfo),
+		taskID2SlotIndex:  make(map[int64]int),
+		executorSlotInfos: make([]*proto.Task, 0),
 		available:         10,
 	}
 
 	var (
 		taskID  = int64(1)
 		taskID2 = int64(2)
+		taskID3 = int64(3)
+		task    = &proto.Task{
+			ID:          taskID,
+			Priority:    1,
+			Concurrency: 1,
+		}
+		task2 = &proto.Task{
+			ID:          taskID2,
+			Priority:    2,
+			Concurrency: 10,
+		}
 	)
 
-	task := &proto.Task{
-		ID:          taskID,
-		Priority:    1,
-		Concurrency: 1,
-	}
-	require.True(t, sm.canAlloc(task))
+	canAlloc, tasksNeedFree := sm.canAlloc(task)
+	require.True(t, canAlloc)
+	require.Nil(t, tasksNeedFree)
 	sm.alloc(task)
-	require.Equal(t, 1, sm.executorSlotInfos[taskID].priority)
-	require.Equal(t, 1, sm.executorSlotInfos[taskID].slotCount)
+	require.Len(t, sm.executorSlotInfos, 1)
+	require.Equal(t, task, sm.executorSlotInfos[sm.taskID2SlotIndex[taskID]])
 	require.Equal(t, 9, sm.available)
 
-	require.False(t, sm.canAlloc(&proto.Task{
-		ID:          taskID2,
-		Priority:    2,
-		Concurrency: 10,
-	}))
+	// the available slots is not enough for task2
+	canAlloc, tasksNeedFree = sm.canAlloc(task2)
+	require.False(t, canAlloc)
+	require.Nil(t, tasksNeedFree)
+
+	// increase the priority of task2, task2 is waiting for allocation
+	task2.Priority = 0
```

Comment:
need a case that there's not enough slots even cancel all lower order tasks
say 3 tasks, 2 lower than current task